src/main.ts

```diff
@@ -62,6 +62,30 @@ function handleError(error: Error) {
   core.setFailed(error.message);
 }
 
+function getGithubHeadRefInfo(): {headRef: string; headSha: string} {
+  const workflowRunPayload = github.context.payload.workflow_run;
+  const pullRequestPayload = github.context.payload.pull_request;
+  const workflowRunPullRequest = workflowRunPayload?.pull_requests?.[0];
+
+  const head_sha =
+    pullRequestPayload?.head.sha ||
+    workflowRunPullRequest?.head.sha ||
+    workflowRunPayload?.head_sha;
+
+  if (typeof head_sha !== 'string') {
+    throw new Error('Could not read head sha');
```

Comment:
Should we throw or just ignore and move on?

Reply:
I think we require this down the line to download artifacts and it was the one that had no fallback, so I assumed it's required. I'm leaning towards throw and we can change if this fails often